tools/gen-random-bout/gen-random-bout.cpp

```diff
@@ -58,80 +69,186 @@ static void push_range(KTest *b, const char *name, unsigned value) {
 
-  for(i = 2; i < (unsigned)argc; i++) {
-    if(strcmp(argv[i], "--sym-args") == 0) {
+  for (i = 2; i < (unsigned)argc; i++) {
+    if (strcmp(argv[i], "--sym-arg") == 0 || strcmp(argv[i], "-sym-arg") == 0) {
+      unsigned nbytes = getint(argv[++i]);
+
+      // A little different than how klee does it but more natural for random.
+      char arg[1024];
+      unsigned x = random() % (nbytes + 1);
+
+      sprintf(arg, "arg%d", total_args++);
+      push_random_obj(&b, arg, x, nbytes + 1);
+    } else if (strcmp(argv[i], "--sym-args") == 0 ||
+               strcmp(argv[i], "-sym-args") == 0) {
       unsigned lb = getint(argv[++i]);
       unsigned ub = getint(argv[++i]);
       unsigned nbytes = getint(argv[++i]);
 
-      narg = random() % (ub - lb) + lb;
-      push_range(&b, "range", narg);
+      if (ub < lb) {
+        fprintf(stderr, "--sym-args first argument should be no more than its "
+                        "second argument");
+        exit(1);
+      }
+
+      narg = random() % (ub - lb + 1) + lb;
+      push_range(&b, "n_args", narg);
 
-      while(narg-- > 0) {
+      while (narg-- > 0) {
         unsigned x = random() % (nbytes + 1);
 
         // A little different than how klee does it but more natural
         // for random.
-        static int total_args;
         char arg[1024];
 
         sprintf(arg, "arg%d", total_args++);
-        push_obj(&b, arg, x, nbytes+1);
-      }
-    } else if(strcmp(argv[i], "--sym-stdout") == 0) {
-      if(!sym_stdout) {
-        sym_stdout = 1;
-        push_obj(&b, "stdout", 1024, 1024);
-        push_obj(&b, "stdout-stat", sizeof(struct stat64), 
-                 sizeof(struct stat64));
+        push_random_obj(&b, arg, x, nbytes + 1);
       }
-    } else if(strcmp(argv[i], "--sym-files") == 0) {
+    } else if (strcmp(argv[i], "--sym-stdout") == 0 ||
+               strcmp(argv[i], "-sym-stdout") == 0) {
+      sym_stdout = 1;
+    } else if (strcmp(argv[i], "--sym-stdin") == 0 ||
+               strcmp(argv[i], "-sym-stdin") == 0) {
+      stdin_size = getint(argv[++i]);
+    } else if (strcmp(argv[i], "--sym-files") == 0 ||
+               strcmp(argv[i], "-sym-files") == 0) {
       unsigned nfiles = getint(argv[++i]);
       unsigned nbytes = getint(argv[++i]);
 
-      while(nfiles-- > 0) {
-        push_obj(&b, "file", nbytes, nbytes);
-        push_obj(&b, "file-stat", sizeof(struct stat64), sizeof(struct stat64));
+      total_files = 0;
+      while (nfiles-- > 0) {
+        file_sizes[total_files++] = nbytes;
       }
 
-      push_obj(&b, "stdin", nbytes, nbytes);
-      push_obj(&b, "stdin-stat", sizeof(struct stat64), sizeof(struct stat64));
     } else {
       fprintf(stderr, "unexpected option <%s>\n", argv[i]);
       assert(0);
     }
   }
 
+  for (i = 0; i < total_files; ++i) {
+    char filename[7] = "A-data";
+    char file_stat[12] = "A-data-stat";
+    unsigned nbytes = file_sizes[i];
+    struct stat64 s;
```

Comment:
`struct stat64 s{};` + other occurences

Reply:
@251 I'll keep this as it is, as I think not including the `{}` is more common in C, assuming this file is C.